src/ofbx.cpp

```diff
@@ -1535,6 +1587,198 @@ struct TextureImpl : Texture
+		return false;
+	}
+
+	bool doesDrawFrontFacingVolumetricLight() const override
+	{
+		// Return the draw front-facing volumetric light property based on the stored data (WIP)
+		return false;
+	}
+
+	Color getColor() const override { return color; }
+	double getIntensity() const override { return intensity; }
+	double getInnerAngle() const override { return innerAngle; }
+	double getOuterAngle() const override { return outerAngle; }
+
+	double getFog() const override { return fog; }
+
+	DecayType getDecayType() const override { return decayType; }
+	double getDecayStart() const override { return decayStart; }
+
+	// Near attenuation
+	bool doesEnableNearAttenuation() const override { return enableNearAttenuation; }
+	double getNearAttenuationStart() const override { return nearAttenuationStart; }
+	double getNearAttenuationEnd() const override { return nearAttenuationEnd; }
+
+	// Far attenuation
+	bool doesEnableFarAttenuation() const override { return enableFarAttenuation; }
+	double getFarAttenuationStart() const override { return farAttenuationStart; }
+	double getFarAttenuationEnd() const override { return farAttenuationEnd; }
+
+	// Shadows
+	const Texture* getShadowTexture() const override { return shadowTexture; }
+	bool doesCastShadows() const override { return castShadows; }
+	Color getShadowColor() const override { return shadowColor; }
+
+	// Area light shape
+	AreaLightShape getAreaLightShape() const override { return areaLightShape; }
+
+	// Barn doors
+	float getLeftBarnDoor() const override { return leftBarnDoor; }
+	float getRightBarnDoor() const override { return rightBarnDoor; }
+	float getTopBarnDoor() const override { return topBarnDoor; }
+	float getBottomBarnDoor() const override { return bottomBarnDoor; }
+	bool doesEnableBarnDoor() const override { return enableBarnDoor; }
+
+	// Member variables to store light properties
+	//-------------------------------------------------------------------------
+	LightType lightType = LightType::POINT; // Light type
+	bool castLight = true;					// Whether the light casts light on objects
+	Color color = {1, 1, 1};					// Light color (RGB values)
+	double intensity = 100.0;				// Light intensity
+
+	// Spotlight properties
+	double innerAngle = 0.0;
+	double outerAngle = 45.0;
+
+	// Light fog intensity
+	double fog = 50;
+
+	// Light decay properties
+	DecayType decayType = DecayType::QUADRATIC;
+	double decayStart = 1.0;
+
+	// Near attenuation properties
+	bool enableNearAttenuation = false;
+	double nearAttenuationStart = 0.0;
+	double nearAttenuationEnd = 0.0;
+
+	// Far attenuation properties
+	bool enableFarAttenuation = false;
+	double farAttenuationStart = 0.0;
+	double farAttenuationEnd = 0.0;
+
+	// Shadow properties
+	const Texture* shadowTexture = nullptr;
+	bool castShadows = true;
+	Color shadowColor = {0, 0, 0};
+
+	// Area light properties
+	AreaLightShape areaLightShape = AreaLightShape::RECTANGLE;
```

Comment:
There's a bunch of of members that are not really set/read from fbx, like this one. Please remove